### PR TITLE
screwdriver_act signal

### PR DESCRIPTION
**Pull request by Garen7**
:cl: Garen
code: adds a signal for screwdriver_act
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

I made this because I was going to run something when attacked by a screwdriver and it felt dumb doing a screwdriver act in attackby.

Most the time I write if(..()) return TRUE but if it ever returns without a value specified I write . = ..() if(.) return TRUE instead.

### CODE REVIEW

**Comment by octareenroon91** [comment]
Does this change anything mechanically, or is that going to be a different PR?

**Comment by Garen7** [comment]
@octareenroon91 nah, the actual functional changes you’d see will be in a later pr

**Comment by ExcessiveUseOfCobblestone** [comment]
i'm very confused.

**Comment by Garen7** [comment]
@ExcessiveUseOfCobblestone about what?

**Comment by Kagus** [comment]
> @ExcessiveUseOfCobblestone about what?

About why the fuck someone thought . and ..() were a good idea when designing DM. Probably.

File: code/game/machinery/camera/camera_assembly.dm
Comment (Jared-Fogle) [inline code/game/machinery/camera/camera_assembly.dm]: You can clean all these up just with `if((. = ..())`, I believe.

**Comment by ExcessiveUseOfCobblestone** [comment]
Why would you need the signal vs just using the proc?

**Comment by SpaceManiac** [comment]
>Why would you need the signal vs just using the proc?

If you want a component to react to it.

A little cleaner than listening for the attackby signal and checking istype.

**Comment by cyclowns** [comment]
seems a little odd to only change one of the tool_act()s to be a signal

**Comment by octareenroon91** [comment]
>seems a little odd to only change one of the tool_act()s to be a signal


Aug 21 17:18:15 (Garen) ok so then yes, i should make the screwdriver_act signal but not the others?\
Aug 21 17:19:05 (Naksu) make the signals you actually use but nothing else

File: code/game/machinery/camera/camera_assembly.dm
Comment (Garen7) [reply code/game/machinery/camera/camera_assembly.dm]: I would do that but I'm worried the syntax would confuse someone so I'm going to hold off on that unless I'm asked to by a maintainer.

Comment (ninjanomnom) [reply code/game/machinery/camera/camera_assembly.dm]: Yeah don't assign vars in conditionals

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**
